### ISSUE TITLE
[Feature]: Allow overwriting keymaps rather than replacing them

### ISSUE THREAD

**Issue by eulertour** [ChristianChiarulli/LunarVim#1140]
### Feature motivation

If you add any keybind to `lvim.keys` for a given mode all of the default keybinds for that mode are removed. I feel that it'd be more intuitive for custom keybinds to extend or overwrite the defaults rather than replace them entirely.

### Describe the expected solution

Extending the default keybinds.

### Describe the alternatives you've considered

Copying all of the default keybinds from `keymappings.lua` into `lv-config.lua`.

**Comment by deryrahman** [comment]
I think it's a good idea. Just looking into [this line](https://github.com/ChristianChiarulli/LunarVim/blob/176106f2234ef7807918b0c131bbd70df3e7c562/lua/keymappings.lua#L99), if the user has assigned the custom keymap on their `lv-config`, for example `lvim.keys.normal_mode = { ... }`, the default keymap would be skipped. Would be more convenient if we're able to override them.

I've already implemented the solution to use map data structure instead of array. So, the configuration on lv-config would be like this
```lua
lvim.keys.normal_mode['po'] = '<ESC>' -- use default opt based on mode
lvim.keys.normal_mode['po'] = {'<ESC>', { silent = true, noremap = true } } -- if you want to use custom opt
```

**Comment by jimcornmell** [comment]
I'm overriding the built-in config for galaxyline completely, see the very end [of this](https://github.com/jimcornmell/LunarVim/blob/master/lv-config.lua) and [this](https://github.com/jimcornmell/LunarVim/blob/master/user.lua).   Is this the correct way to do it?

**Comment by abzcoding** [comment]
Fixed via #1176
